scripts/externalTests/common.sh

```diff
@@ -106,25 +106,31 @@ function setup_solc
 
 function download_project
 {
-    local repo="$1"
+    local repo_url="$1"
     local ref_type="$2"
-    local solcjs_ref="$3"
+    local ref="$3"
     local test_dir="$4"
 
-    [[ $ref_type == commit || $ref_type == branch || $ref_type == tag ]] || assertFail
+    [[ $ref_type == commit || $ref_type == branch || $ref_type == tag || $ref_type == release ]] || assertFail
 
-    printLog "Cloning ${ref_type} ${solcjs_ref} of ${repo}..."
-    if [[ $ref_type == commit ]]; then
-        mkdir ext
-        cd ext
-        git init
-        git remote add origin "$repo"
-        git fetch --depth 1 origin "$solcjs_ref"
-        git reset --hard FETCH_HEAD
-    else
-        git clone --depth 1 "$repo" -b "$solcjs_ref" "$test_dir/ext"
-        cd ext
+    printLog "Cloning ${repo_url}..."
+    # Clone the repo ignoring all blobs until needed by git.
+    # This allows access to commit history but with a fast initial clone
+    git clone --filter=blob:none "$repo_url" "$test_dir/ext"
```

Comment:
Nice. I was going to comment about dropping `--depth 1`, potentially making the clone longer, but looks like this would make it a non-issue?